test/blockedLinks.js

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+describe("Blocked Links API", function() {
+  "use strict";
+
+  var firstLink = {url: "http://example0.com/", title: "site#0"};
+  var secondLink = {url: "http://example1.com/", title: "site#1"};
+  var thirdLink = {url: "http://example2.com/", title: "site#2"};
+
+  afterEach(function(done) {
+    // Clear the database and cached links.
```

Comment:
So I tried this out and it seems to change the scope such that the calls to `init()` and `reset()` in the `then()` function need to directly reference `gBlockedLinks` because `this` is `undefined` from there perspective. Is that expected?